packages/grafana-runtime/src/components/DataSourcePicker.tsx

```diff
@@ -80,7 +81,14 @@ export class DataSourcePicker extends PureComponent<DataSourcePickerProps, DataS
     }
   }
 
-  onChange = (item: SelectableValue<string>) => {
+  onChange = (item: SelectableValue<string>, actionMeta: ActionMeta) => {
+    if (actionMeta.action === 'clear') {
+      if (typeof this.props.onClear === 'function') {
```

Comment:
Not really, I think TypeScript would suffice to make sure we're really passing a function here – I can do a simple truthiness check and that would make this condition a bit clearer because we could combine the `action` check with the prop check 👍